pkg/config/images.go

```diff
@@ -19,16 +18,16 @@ func openShiftVersion(imageVersion string) (string, error) {
 	return fmt.Sprintf("v%s.%s.%s", parts[0][:1], parts[0][1:], parts[1]), nil
 }
 
-func selectNodeImage(cs *api.OpenShiftManagedCluster, deployOS string) {
+func selectNodeImage(cs *api.OpenShiftManagedCluster, pluginConfig *api.PluginConfig) {
```

Comment:
this works but is ugly.  As mentioned before I'd like to see all these functions that take `cs *api.OpenShiftManagedCluster, pluginConfig *api.PluginConfig` being methods on a base struct which contains cs and pluginConfig as fields.  That way they don't need to be passed around everywhere.

Reply:
@thekad I think I didn't explain myself well enough here.  In pkg/config (not pkg/api) create a new unexported struct type just like in the other packages.  That should contain cs *api.OpenShiftManagedCluster and pluginConfig *api.PluginConfig.  Then all of these functions here should be member functions of that struct type.

Reply:
@kargakis fyi ^